### DISCUSSION TITLE
Using jsWorker under Metro (react native expo project)

### DISCUSSION THREAD

**Discussion by tvbegovic** [ujjwalguptaofficial/JsStore#363] [Q&A]
Hi,
I migrated my legacy react native app to new Expo 50 that no longer uses webpack but Metro. The example code to get web worker path that was working under webpack, no longer works in Metro.
```
const getWorkerPath = () => {

    // return dev build when env is development
    if (process.env.NODE_ENV === 'development') {
        
        return require("file-loader?name=scripts/[name].[hash].js!jsstore/dist/jsstore.worker.js");

    }
    else { // return prod build when env is production
        
        return require("file-loader?name=scripts/[name].[hash].js!jsstore/dist/jsstore.worker.min.js");
        
    }
};
```
Is there a way to initialize jsstore connection using workers under Metro or my only option is to avoid using workers?

**Comment by ujjwalguptaofficial** [comment]
i will have a look for how to do it in metro.

But you can use the web worker without the bundler support. Just copy paste the worker file into a static directly like where you have images and use directly in the Worker constructor.

```
new Worker(`ff`);
```

**Comment by ujjwalguptaofficial** [comment]
Here is a metro example - 

```
const getWorkerPath = () => {
    // return dev build when env is development
    if (__DEV__) {
        return require("jsstore/dist/jsstore.worker.js");
    } else { // return prod build when env is production
        return require("jsstore/dist/jsstore.worker.min.js");
    }
};

export default getWorkerPath;

```

I have not tested it, please test and confirm.
